cmd/soroban-rpc/lib/preflight/src/lib.rs

```diff
@@ -2,30 +2,48 @@ extern crate anyhow;
 extern crate base64;
 extern crate libc;
 extern crate sha2;
-extern crate soroban_env_host;
-extern crate soroban_simulation;
-
-use anyhow::{anyhow, bail, Result};
-use sha2::{Digest, Sha256};
-use soroban_env_host::storage::EntryWithLiveUntil;
-use soroban_env_host::xdr::{
-    AccountId, ExtendFootprintTtlOp, Hash, InvokeHostFunctionOp, LedgerEntry, LedgerEntryData,
-    LedgerFootprint, LedgerKey, LedgerKeyTtl, OperationBody, ReadXdr, ScErrorCode, ScErrorType,
-    SorobanTransactionData, TtlEntry, WriteXdr,
-};
-use soroban_env_host::{HostError, LedgerInfo, DEFAULT_XDR_RW_LIMITS};
-use soroban_simulation::simulation::{
-    simulate_extend_ttl_op, simulate_invoke_host_function_op, simulate_restore_op,
-    InvokeHostFunctionSimulationResult, LedgerEntryDiff, RestoreOpSimulationResult,
-    SimulationAdjustmentConfig,
-};
-use soroban_simulation::{AutoRestoringSnapshotSource, NetworkConfig, SnapshotSourceWithArchive};
+extern crate soroban_env_host_curr;
+extern crate soroban_env_host_prev;
+extern crate soroban_simulation_curr;
+extern crate soroban_simulation_prev;
+
+pub(crate) use anyhow::{anyhow, bail, Result};
+pub(crate) use sha2::{Digest, Sha256};
+
+// We support two different versions of soroban simutlaneously, switching on the
+// protocol version each supports. This is the exact same mechanism we use in
+// stellar-core to switch soroban hosts on protocol boundaries, and allows
+// synchronously cutting over between significantly different versions of the
+// host (or VM) without having to do fine-grained versioning within the VM.
+//
+// The way it is _accomplished_ is by mounting the same adaptor code (in
+// `shared.rs`) at two different paths in the module tree, and then providing
```

Comment:
I don't see `shared.rs` file in this PR, did you forget to commit it?

Reply:
Oh, oops, yes. Added, sorry!

Reply:
LGTM, thanks.